### PR TITLE
feat(multicluster) client for multicluster config

### PR DESCRIPTION
**Pull request by timmyreilly**
Signed-off-by: timmyreilly <tireilly@microsoft.com>

Adding the config client for multicluster for #3456 and #3449

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [X] |
| Documentation              | [ ] |
| Install                    | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Ingress                    | [ ] |
| Egress                     | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| SMI Policy                 | [ ] |
| Sidecar Injection          | [ ] |
| Security                   | [ ] |
| Upgrade                    | [ ] |
| Tests                      | [ ] |
| CI System                  | [ ] |
| Demo                       | [ ] |
| Performance                | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?

1. Is this a breaking change?

### CODE REVIEW

File: pkg/config/types.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+type client struct {
+	informers      *informerCollection
+	caches         *cacheCollection
+	cacheSynced    chan interface{}
+	kubeController kubernetes.Controller
+}
+
+// Controller is the interface for the functionality provided by the resources part of the multiclusterservice.openservicemesh.io API group
+type Controller interface {
+	// TODO: specify required functions
```
Comment (steeling) [inline pkg/config/types.go:35]: It's get and list

File: pkg/config/types.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	"github.com/openservicemesh/osm/pkg/logger"
+)
+
+var (
+	log = logger.New("multiclusterservice-controller")
+)
+
+// informerCollection is the type used to represent the collection of informers for the multiclusterservice.openservicemesh.io API group
+type informerCollection struct {
+	remoteService cache.SharedIndexInformer
```
Comment (steeling) [inline pkg/config/types.go:17]: multiclusterService

File: pkg/config/types.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+// cacheCollection is the type used to represent the collection of caches for the multiclusterservice.openservicemesh.io API group
+type cacheCollection struct {
+	remoteService cache.Store
```
Comment (steeling) [inline pkg/config/types.go:22]: same

File: pkg/config/types.go (hunk 1 above)
Comment (timmyreilly) [reply pkg/config/types.go:35]: GetMulticlusterService? and ListMulticlusterService?

File: pkg/config/client.go [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+	var services []*v1alpha1.MultiClusterService
+
+	for _, mcs := range c.caches.multiClusterService.List() {
+		mcservice := mcs.(*v1alpha1.MultiClusterService)
+		services = append(services, mcservice)
+	}
+	return services
+}
+
+func (c client) GetMultiClusterServices(source identity.K8sServiceAccount) []*v1alpha1.MultiClusterService {
```
Comment (steeling) [inline pkg/config/client.go:107]:
```suggestion
func (c client) GetMultiClusterService(name, namespace string) []*v1alpha1.MultiClusterService {
```

File: pkg/config/types.go (hunk 1 above)
Comment (steeling) [reply pkg/config/types.go:37]: yup, see above comment

File: pkg/config/client.go (hunk 4 above)
Comment (timmyreilly) [reply pkg/config/client.go]: I took another pass

File: pkg/config/client.go [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+		mcservice := mcs.(*v1alpha1.MultiClusterService)
+		services = append(services, mcservice)
+	}
+	return services
+}
+
+func (c client) GetMultiClusterServices(name, namespace string) []*v1alpha1.MultiClusterService {
+	var services []*v1alpha1.MultiClusterService
+
+	for _, mcs := range c.caches.multiClusterService.List() {
```
Comment (steeling) [inline pkg/config/client.go:109]:
```suggestion
	for _, ob := range c.caches.multiClusterService.List() {
	    mcs := obj.(*v1alpha1.MultiClusterService)
```

File: pkg/config/types.go [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+	caches         *cacheCollection
+	cacheSynced    chan interface{}
+	kubeController kubernetes.Controller
+}
+
+// Controller is the interface for the functionality provided by the resources part of the multiclusterservice.openservicemesh.io API group
+type Controller interface {
+	// TODO: specify required functions
+	ListMultiClusterServices() []*v1alpha1.MultiClusterService
+	GetMultiClusterServices(name, namespace string) []*v1alpha1.MultiClusterService
```
Comment (steeling) [inline pkg/config/types.go:38]: GetMultiClusterService (singular)

**Review by snehachhabria — APPROVED** [review]

**Review by shashankram — APPROVED** [review]

**Pull request merged**